src/app/loadout/mod-utils.ts

```diff
@@ -102,14 +102,14 @@ export function createGetModRenderKey() {
  */
 export function getItemEnergyType(
   item: DimItem,
-  lockArmorEnergyType: LockArmorEnergyType | undefined,
+  armorEnergyRules: ArmorEnergyRules,
```

Comment:
I'm not a huge fan of all these functions knowing about all the energy rules when they don't need all that information. It's only really a nit though so happy to let you decide.